orttraining/orttraining/python/training/ortmodule/_graph_execution_manager.py

```diff
@@ -234,25 +238,69 @@ def _get_exported_model(self, *inputs, **kwargs):
         # correct training flag to reflect the expected behavior.
         # For example, the Dropout node in a model is dropped under eval mode.
         assert self._export_mode is not None, "Please use a concrete instance of ExecutionManager"
-
+        exported_model = None
         try:
-            with torch.no_grad(), _logger.suppress_os_stream_output(log_level=self._loglevel):
-                torch.onnx.export(self._flattened_module,
-                                  sample_inputs_as_tuple,
-                                  f,
-                                  input_names=self._input_info.names,
-                                  output_names=output_names,
-                                  opset_version=ONNX_OPSET_VERSION,
-                                  do_constant_folding=False,
-                                  training=self._export_mode,
-                                  dynamic_axes=self._input_info.dynamic_axes,
-                                  verbose=self._loglevel < _logger.LogLevel.WARNING,
-                                  export_params=False,
-                                  keep_initializers_as_inputs=True)
+            if self._onnx_export_type == torch.onnx.OperatorExportTypes.ONNX_FALLTHROUGH:
+                cloned_flattened_module = copy.deepcopy(self._flattened_module)
```

Comment:
This is inspired by @wschin. There is an explanation in the mail let me copy it here. 

Thiago :
"For the deep copy comment, we perform a deep copy of the model because some models can be stateful. This is an issue because the very fact that pytorch exporter runs the user model to get the onnx graph (through tracing) can possibly modify the model, so the first “real forward run” (not the onnx export run) would start on the wrong state. We first detected this issue when working on ORTTrainer frontend, but the same principle applies to ORTModule as both rely on ONNX tracing"

[pengwa] For the deep copy before exporting, Thanks Thiago for the detailed explanations. Just want surface that line is newly added in our PR, we not only copied the model but also removed torch.no_grad() to make sure in exported graph we have correct requires_grad attributes. I